src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Configuration/BaseProjectConfigurationDimensionProvider.cs

```diff
@@ -205,6 +230,28 @@ async Task UpdateUnderLockAsync(UnconfiguredProject project, Action<ProjectRootE
             }
         }
 
+        public virtual async Task<IEnumerable<KeyValuePair<string, IEnumerable<string>>>> GetProjectConfigurationDimensionsAsync(Project project)
+        {
+            Requires.NotNull(project, nameof(project));
+
+            ImmutableArray<string> values = GetOrderedPropertyValues(project);
+            if (values.IsEmpty)
+            {
+                return ImmutableArray<KeyValuePair<string, IEnumerable<string>>>.Empty;
```

Comment:
Minor, but this and other similar code in this class could use `Enumerable.Empty` to reduce allocations a little.

```suggestion
                return Enumerable.Empty<KeyValuePair<string, IEnumerable<string>>>();
```